presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -340,6 +340,23 @@ public void insertPartitionedTable(HiveStorageFormat storageFormat)
     }
 
     @Test
+    public void testDeleteFromUnpartitionedTable()
+            throws Exception
+    {
+        @Language("SQL") String createTable = "CREATE TABLE test_delete_unpartitioned AS SELECT * FROM (VALUES (1, 'a'), (2, 'b')) t(_bigint, _varchar)";
+
+        assertQuery(createTable, "SELECT 2");
+
+        try {
+            queryRunner.execute("DELETE FROM test_delete_unpartitioned");
+            fail("expected exception");
+        }
+        catch (RuntimeException e) {
+            assertEquals(e.getMessage(), "This connector only supports delete where one or more partitions are deleted entirely");
```

Comment:
Can you check the error code here too?

`HiveMetadata` will throw the wrong exception because it uses `UnsupportedOperationException`. It should be `PrestoException(NOT_SUPPORTED)`